src/spanningtrees/boruvka.jl

```diff
@@ -0,0 +1,56 @@
+@traitfn function boruvka_mst(graph::AG::(!IsDirected), 
+    distmx::AbstractMatrix{T} = weights(graph); 
+    minimize = true) where {T<:Real,U,AG<:AbstractGraph{U}}
+    
+    f_distmx = minimize ? copy(distmx) : copy(-distmx)
```

Comment:
this can make things very expensive. Is there another way to do this without a copy of a weights matrix? This becomes a dense matrix for `defaultdistance`.

Reply:
> this can make things very expensive. Is there another way to do this without a copy of a weights matrix? This becomes a dense matrix for `defaultdistance`.

Yeah! you're right I did it cause it was kinda easy to the comparison in case minimize = false, but I can do it much more efficiently, I'm sorry, I'm gonna update it later and it will be much more efficient, but thanks for the observation!

Reply:
Just to be clear: we'll want the change prior to merge. We generally don't accept enhancements that only work on small graphs (except in `Experimental` and sometimes in rare cases), and you won't get much farther than 1e6 - 1e7 with this.

Reply:
This is holding things up at this point. Copying the distmx is a bad idea.

Reply:
@sbromberger  in the last commit I`m not copying it anymore, this is outdated.